### ISSUE TITLE
Naming conflict in const Model and interface Model

### ISSUE THREAD

**Issue by achtan** [Automattic/mongoose#12873]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.8.2

### Node.js version

16.8

### MongoDB server version

5.1

### Typescript version (if applicable)

4.9.4

### Description

from models.d.ts we are exporting `const Model` and also `interface Model` and my IDE (WebStrom) is confused about what I want to import, even when I use the `import type` syntax:

### Steps to Reproduce

![Screenshot 2023-01-04 at 21 07 52](https://user-images.githubusercontent.com/1048212/210640607-abc8c110-5bfb-4fed-84f8-33dd43f11d34.png)

as you can see MM refers to both (const and interface) and because of this, I have other issues with suggestions etc. in my IDE...

for example: 
![Screenshot 2023-01-04 at 21 27 09](https://user-images.githubusercontent.com/1048212/210643833-9816bfc3-2903-4375-94b1-ae97568eccca.png)
IDE is not sure about `MM`.

note: this 
```ts
export default (models.LicenseModel as MM<License>) ||
  model<License>('LicenseModel', Schema, 'licenses')
```
is there because of the hot module issue

PS: the `/** @type {import('mongoose').Model} */` don't help either.

### Expected Behavior

Can we rename `const Model` or `interface Model` to something else?

**Comment by vkarpov15** [comment]
Fixed by #12935

**Comment by vkarpov15** [comment]
On further review, we're going to undo this change and label this issue as a won't fix. If we remove `const Model`, we lose support for `obj instanceof mongoose.Model` in TypeScript, which is not something we want to do. It is unfortunate that this syntax causes IDE issues, but I think that is better than eliminating valid functionality.
